src/CattyTests/FileManager/CBFileManagerTests.swift

```diff
@@ -121,4 +121,113 @@ final class CBFileManagerTests: XCTestCase {
+        }
+
+        wait(for: [expectation], timeout: 1.0)
+    }
+
+    func testLoadPreviewImageAndCacheWhenAutomaticScreenshotNotCached() {
+        let info = ProjectLoadingInfo.init(forProjectWithName: kDefaultProjectBundleName, projectID: kNoProjectIDYetPlaceholder)!
+        let project = Project(loadingInfo: info)!
+
+        let automaticScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotAutoFilename
+        let automaticScreenshot = UIImage(color: UIColor.red)!
+
+        let imageCache = RuntimeImageCacheMock(thumbnails: [automaticScreenshotThumbnailPath: automaticScreenshot], cachedImages: [:])
+        let filePath = project.projectPath() + kScreenshotAutoFilename
+
+        let fileManager = CBFileManagerMock(imageCache: imageCache, filePath: [filePath])
+
+        let expectation = XCTestExpectation(description: "Load image from disk - Automatic Screenshot")
+
+        print("Running: testLoadPreviewImageAndCacheWhenAutomaticScreenshotNotCached")
+        _ = fileManager.loadPreviewImageAndCache(projectLoadingInfo: info) { image, path in
+            XCTAssertEqual(automaticScreenshotThumbnailPath, path)
+            XCTAssertEqual(automaticScreenshot, image)
+
+            expectation.fulfill()
+        }
+
+        wait(for: [expectation], timeout: 1.0)
+    }
+
+    func testLoadPreviewImageAndCacheWhenManualScreenshotNotCached() {
+        let info = ProjectLoadingInfo.init(forProjectWithName: kDefaultProjectBundleName, projectID: kNoProjectIDYetPlaceholder)!
+        let project = Project(loadingInfo: info)!
+
+        let manualScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotManualFilename
+        let manualScreenshot = UIImage(color: UIColor.green)!
+
+        let automaticScreenshotThumbnailPath = project.projectPath() + kScreenshotThumbnailPrefix + kScreenshotAutoFilename
+        let automaticScreenshot = UIImage(color: UIColor.red)!
+
+        let imageCache = RuntimeImageCacheMock(thumbnails: [automaticScreenshotThumbnailPath: automaticScreenshot, manualScreenshotThumbnailPath: manualScreenshot], cachedImages: [:])
+
+        let autoScreenshotFilePath = project.projectPath() + kScreenshotAutoFilename
+        let manualScreenshotFilePath = project.projectPath() + kScreenshotManualFilename
+
+        let fileManager = CBFileManagerMock(imageCache: imageCache, filePath: [autoScreenshotFilePath, manualScreenshotFilePath])
+
+        let expectation = XCTestExpectation(description: "Load image from disk - Manual Screenshot")
+
+        print("Running: testLoadPreviewImageAndCacheWhenAutomaticScreenshotNotCached")
+        _ = fileManager.loadPreviewImageAndCache(projectLoadingInfo: info) { image, path in
+            XCTAssertEqual(manualScreenshotThumbnailPath, path)
+            XCTAssertEqual(manualScreenshot, image)
+
+            expectation.fulfill()
+        }
+
+        wait(for: [expectation], timeout: 1.0)
```

Comment:
Please add an additional test to verify what is happening when there is not screenshot found.